debian/additions/debian-start.inc.sh

```diff
@@ -21,26 +21,38 @@ function check_for_crashed_tables() {
   # spaces in the thing to be looped over.
 
   # If a crashed table is encountered, the "mariadb" command will return with a status different from 0
+  #
+  # The first query will generate lines like.
+  #   select count(*) into @discard from 'mysql'.'db'
+  # The second line will load all tables without printing any actual results,
+  # but may show warnings and definitely is expected to have some error and
+  # exit code if crashed tables are encountered.
+  #
+  # Note that inside single quotes must be quoted with '\'' (to be outside of single quotes).
   set +e
-
-  LC_ALL=C $MARIADB --skip-column-names --batch -e  '
-      select concat('\''select count(*) into @discard from `'\'',
-                    TABLE_SCHEMA, '\''`.`'\'', TABLE_NAME, '\''`'\'')
-      from information_schema.TABLES where TABLE_SCHEMA<>'\''INFORMATION_SCHEMA'\'' and TABLE_SCHEMA<>'\''PERFORMANCE_SCHEMA'\'' and ( ENGINE='\''MyISAM'\'' or ENGINE='\''Aria'\'' )' | \
-    xargs -i ${MARIADB} --skip-column-names --silent --batch \
-                    --force -e "{}" &>"${tempfile}"
+  # The $MARIADB is intentionally used to expand into a command and arguments
+  # shellcheck disable=SC2086
+  LC_ALL=C echo '
```

Comment:
At the risk of being too pedantic - beginning of this function says "trigging myisam-recover" and it doesn't. So minimally a better comment there. Somehow a basic open of the table is sufficient - was there a decision recorded for this (its a big search - I couldn't find it)? [check table {} FAST](https://mariadb.com/kb/en/check-table/) could have been used as SQL.

Also (for later), looking for filesystem based files might be quicker than information_schema. And Aria might auto-recover (especially after 10.6).

`LC_ALL=C` is now on echo (which serves no purpose) rather than `$MARIADB` which only effects output into the email. So I'm not sure there's a purpose in including it at all.

SC2086 is due to `$MARIADB` being unquoted (twice now which could be corrected.

`xargs --no-run-if-empty` would be worth saving a needless execution.

Is xargs `-P` desired to perform concurrent

Reply:
reminder ^

Reply:
Checked with Monty - Aria tables will auto recover. MyISAM won't.
Option 1:
* remove all between set +e / -e as old stuff assuming we never need to handle crashed upgrade from < 10.4, maybe users can handle this if it does happen.
Option 2:
execute as a single SQL string:
```
select concat('check table ', group_concat(CONCAT('`', TABLE_SCHEMA, '`.`', TABLE_NAME, '`')), ' fast') into @check FROM information_schema.TABLES WHERE TABLE_SCHEMA<>"INFORMATION_SCHEMA" AND TABLE_SCHEMA<>'PERFORMANCE_SCHEMA'  AND ENGINE='MyISAM' OR (TABLE_SCHEMA='mysql' and TABLE_NAME='time_zone_leap_second'); execute immediate @check;
``` 

rational:
* check table fast - doesn't need to fully read the entire table (unsure if information_schema will open table)
* MyISAM only is needed
* mysql.time_zone_leap_second include to ensure a single table is there.
* quoting - in case of some table names, isn't fully coverage